package j10_exception;

public class Ex04_MessageNaN {

	public static void main(String[] args) {

		// ** by Zero ( 나누기 0 )
		// => 정수형 연산 : ArithmeticException
		// => 실수형 연산 : Exception 이 일어나지 않음
		// Wrapper 클래스의 메서드 활용

		try {
			double d1 = 1.5 / 0.0; // => Infinity
			double d2 = 1.5 % 0.0; // => NaN
			// => 그러므로 Exception 으로 대응할 수 없음

			if (Double.isInfinite(d1) || Double.isNaN(d2)) {

			} else {
				System.out.printf("실수 by zero Test : d1 = %f, d2 = %f, d1 * 100 = %f\n", d1, d2, (d1 * 100));
			}
		} catch (Exception e) {
			System.out.println("** Exception => " + e.toString());
		} finally {
			System.out.println("** finally_무조건 실행 **");
		}
		System.out.println("** Program 정상 종료 **");
	}// main

}// class